In [1]:
import pandas as pd

In [2]:
import pandas as pd

# Creating a dataset
data = {
    "order_id": [201, 202, 203, 204, 205, 206, 207, 208, 209, 210],
    "product": [
        "Laptop",
        "Smartphone",
        "Laptop",
        "Tablet",
        "Smartphone",
        "Tablet",
        "Smartwatch",
        "Laptop",
        "Headphones",
        "Smartwatch",
    ],
    "category": [
        "Electronics",
        "Electronics",
        "Electronics",
        "Electronics",
        "Electronics",
        "Electronics",
        "Wearables",
        "Electronics",
        "Accessories",
        "Wearables",
    ],
    "region": [
        "North",
        "South",
        "East",
        "North",
        "West",
        "South",
        "East",
        "West",
        "North",
        "West",
    ],
    "sales": [1500, 900, 1200, 600, 750, 450, 320, 1400, 200, 380],
}

df = pd.DataFrame(data)
print(df)


   order_id     product     category region  sales
0       201      Laptop  Electronics  North   1500
1       202  Smartphone  Electronics  South    900
2       203      Laptop  Electronics   East   1200
3       204      Tablet  Electronics  North    600
4       205  Smartphone  Electronics   West    750
5       206      Tablet  Electronics  South    450
6       207  Smartwatch    Wearables   East    320
7       208      Laptop  Electronics   West   1400
8       209  Headphones  Accessories  North    200
9       210  Smartwatch    Wearables   West    380


# ***1.***

***Створи зведену таблицю, яка показує загальні продажі (sales) для кожної категорії продукту (category). Яка категорія має найбільші загальні продажі? Запиши відповідь у текстовому блоці.
.***

In [3]:
total_sum_by_category = pd.pivot_table(df, values="sales", index="category", aggfunc="sum")
total_sum_by_category.sort_values(by="sales", ascending=False).reset_index()

category  sales
0  Electronics   6800
1    Wearables    700
2  Accessories    200

***Категорія Electronics має найбільші загальні продажі***

# ***2.***


***Створи зведену таблицю, яка показує загальні продажі для кожного продукту (product) у кожному регіоні (region). Який продукт мав найвищі продажі в регіоні North? Запиши відповідь у текстовому блоці.***

In [4]:
total_sum_by_product_region = pd.pivot_table(df, values="sales", index=["product", "region"], aggfunc="sum")
total_sum_by_product_region.sort_values(by="sales", ascending=False).reset_index()

product region  sales
0      Laptop  North   1500
1      Laptop   West   1400
2      Laptop   East   1200
3  Smartphone  South    900
4  Smartphone   West    750
5      Tablet  North    600
6      Tablet  South    450
7  Smartwatch   West    380
8  Smartwatch   East    320
9  Headphones  North    200

***Laptop мав найвищі продажі в регіоні North***

# ***3.***


***Створи зведену таблицю, яка показує загальні продажі та середні продажі для кожної категорії продукту. Яка середня сума продажів для категорії Electronics? Запиши відповідь у текстовому блоці.***

In [5]:
total_sum__mean_by_category = pd.pivot_table(df, values="sales", index="category", aggfunc=["sum", "mean"])
total_sum__mean_by_category.sort_values(by=("sum", "sales"), ascending=False).reset_index()

category   sum        mean
               sales       sales
0  Electronics  6800  971.428571
1    Wearables   700  350.000000
2  Accessories   200  200.000000

***Середня сума продажів для категорії Electronics – 971.428571***




# **4.**

***Створи зведену таблицю з використанням параметра fill_value=0, яка показує загальні продажі для кожного продукту та регіону.***


In [6]:
fill_value_by_product_region = pd.pivot_table(df, values="sales", index="product", columns="region", aggfunc="sum",fill_value=0)
fill_value_by_product_region

region      East  North  South  West
product                             
Headphones     0    200      0     0
Laptop      1200   1500      0  1400
Smartphone     0      0    900   750
Smartwatch   320      0      0   380
Tablet         0    600    450     0